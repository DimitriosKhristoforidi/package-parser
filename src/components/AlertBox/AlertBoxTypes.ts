export interface IAlertBox {
  content: string;
}
