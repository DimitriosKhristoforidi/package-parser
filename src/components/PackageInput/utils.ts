import API from '../../API';
import { Modal } from 'antd';
import { TGetPackagesData } from './PackageInputTypes';

export const getPackagesData: TGetPackagesData = async (dependenciesList) => {
  if (!dependenciesList.length) {
    Modal.error({ title: 'JSON is empty' });
    return [];
  }

  const promises = dependenciesList.map((dep) => API.getPackage(dep));

  return await Promise.all(promises).then((res) =>
    res.map((item) => item.results[0].package),
  );
};
