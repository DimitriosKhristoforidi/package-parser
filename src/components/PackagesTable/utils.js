import { Modal, Tag } from 'antd';
import API from '../../API';

export const columns = [
  {
    title: 'Title',
    dataIndex: 'name',
    key: 'name',
    width: '15%',
  },
  {
    title: 'Description',
    dataIndex: 'description',
    key: 'description',
    width: '40%',
  },
  {
    title: 'Tags',
    dataIndex: 'keywords',
    key: 'keywords',
    render: keywords => (
      <>
        {
          keywords?.map(tag =>
            <Tag color="blue" key={tag}>
              {tag.toUpperCase()}
            </Tag>,
          )
        }
      </>
    ),
  },
  {
    title: 'NPM link',
    dataIndex: 'links',
    key: 'links',
    render: links => (
      <a target="_blank" rel="noreferrer" href={links.npm}>{links.npm}</a>
    ),
  },
];

export const getPackagesData = async (list) => {
  if (list.length) {
    const promises = [];
    for (let i in list)
      promises.push(API.getPackage(list[i]));
    return await Promise.all(promises)
      .then(r => r.map(item => item.results[0].package));
  } else {
    Modal.error({ title: 'JSON is empty' });
  }
};
