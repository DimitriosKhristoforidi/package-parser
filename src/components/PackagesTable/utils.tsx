import React from 'react';
import { Tag } from 'antd';

export const columns = [
  {
    title: 'Title',
    dataIndex: 'name',
    key: 'name',
    width: '15%',
  },
  {
    title: 'Description',
    dataIndex: 'description',
    key: 'description',
    width: '40%',
  },
  {
    title: 'Tags',
    dataIndex: 'keywords',
    key: 'keywords',
    render: (keywords: string[]) => (
      <>
        {keywords?.map((keyword, id) => (
          <Tag color='blue' key={`${keyword}-${id}`}>
            {keyword.toUpperCase()}
          </Tag>
        ))}
      </>
    ),
  },
  {
    title: 'NPM link',
    dataIndex: 'links',
    key: 'links',
    render: (links: { npm: string }) => (
      <a target='_blank' rel='noreferrer' href={links.npm}>
        {links.npm}
      </a>
    ),
  },
];
