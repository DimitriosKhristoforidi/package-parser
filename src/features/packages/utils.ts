import {
  TGetEmptyPackage,
  TGetPackagesData,
  TIsArray,
  TIsEmpty,
  TValidatePackagesTerm,
} from './packagesSliceTypes';
import { EMPTY_JSON, ERROR_MESSAGES } from './constants';
import API from '../../API';

export const getEmptyPackage: TGetEmptyPackage = (name) => {
  return {
    name,
    description: '-',
    keywords: [],
    links: [],
  };
};

export const getPackagesData: TGetPackagesData = async (dependenciesList) => {
  const promises = dependenciesList.map((dep) => API.getPackage(dep));

  return await Promise.all(promises).then((res) =>
    res.map(
      (item, idx) =>
        item.results[0]?.package || getEmptyPackage(dependenciesList[idx]),
    ),
  );
};

const isArray: TIsArray = (array) => {
  if (Array.isArray(array)) {
    throw new Error();
  }
};

const isEmpty: TIsEmpty = (line) => {
  if (!line || line === EMPTY_JSON) {
    throw new Error(ERROR_MESSAGES.EMPTY_JSON);
  }
};

export const validatePackagesTerm: TValidatePackagesTerm = async (
  packagesTerm,
) => {
  isEmpty(packagesTerm);
  try {
    const json = JSON.parse(packagesTerm);
    isArray(json);
  } catch {
    throw new Error(ERROR_MESSAGES.NOT_A_JSON);
  }
};
