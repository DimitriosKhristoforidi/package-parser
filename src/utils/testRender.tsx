import React from 'react';
import { render as rtlRender, RenderOptions } from '@testing-library/react';
import { Store } from '@reduxjs/toolkit';
import configureStore from 'redux-mock-store';
import { RootState } from '../app/types';
import { Provider } from 'react-redux';
import { initialStateMock } from '../mock/testMocks';

export const beforeEachTest = () => {
  Object.defineProperty(window, 'matchMedia', {
    writable: true,
    value: jest.fn().mockImplementation((query) => ({
      matches: false,
      media: query,
      onchange: null,
      addListener: jest.fn(),
      removeListener: jest.fn(),
      addEventListener: jest.fn(),
      removeEventListener: jest.fn(),
      dispatchEvent: jest.fn(),
    })),
  });
};

interface ExtendedRenderOptions extends RenderOptions {
  initialState: Partial<RootState>;
  store?: Store<Partial<RootState>>;
}

const render = (
  component: React.ReactElement,
  {
    initialState,
    store = configureStore<Partial<RootState>>()(initialState),
    ...renderOptions
  }: ExtendedRenderOptions = {
    initialState: initialStateMock,
  },
) => {
  return rtlRender(component, {
    wrapper: TestWrapper(store),
    ...renderOptions,
  });
};

const TestWrapper =
  (store: Store) =>
  ({ children }: { children?: React.ReactNode }) => {
    return <Provider store={store}>{children}</Provider>;
  };

export * from '@testing-library/react';
export { render };
