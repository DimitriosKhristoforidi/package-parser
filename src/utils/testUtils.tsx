import { configureStore } from '@reduxjs/toolkit';
import packageReducer from '../features/packages/packagesSlice';
import { render, RenderResult } from '@testing-library/react';
import { Provider } from 'react-redux';
import React, { ReactElement } from 'react';

type TRenderWithRedux = (component: ReactElement) => RenderResult;

export const renderWithRedux: TRenderWithRedux = (component) => {
  const store = configureStore({
    reducer: {
      packages: packageReducer,
    },
  });

  return render(<Provider store={store}>{component}</Provider>);
};
