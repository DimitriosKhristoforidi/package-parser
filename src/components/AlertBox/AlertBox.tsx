import React from 'react';
import { Alert, AlertTitle } from '@mui/material';
import { IAlertBox } from './AlertBoxTypes';

const AlertBox: React.FC<IAlertBox> = ({ content }) => {
  return (
    <div>
      <Alert severity='warning'>
        <AlertTitle>Warning</AlertTitle>
        {content}
      </Alert>
    </div>
  );
};

export default AlertBox;
