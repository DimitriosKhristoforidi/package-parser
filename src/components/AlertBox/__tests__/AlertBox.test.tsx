import React from 'react';
import { renderWithRedux } from '../../../utils/testUtils';
import AlertBox from '../AlertBox';

describe('Alert Box', () => {
  it('Should match snapshot', () => {
    const wrapper = renderWithRedux(<AlertBox content='Content' />);

    expect(wrapper).toMatchSnapshot();
  });
});
