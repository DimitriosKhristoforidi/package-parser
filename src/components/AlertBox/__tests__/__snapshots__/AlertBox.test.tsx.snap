// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert Box Should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div>
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardWarning MuiAlert-standard css-rob5aa-MuiPaper-root-MuiAlert-root"
          role="alert"
        >
          <div
            class="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
              data-testid="ReportProblemOutlinedIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 5.99L19.53 19H4.47L12 5.99M12 2L1 21h22L12 2zm1 14h-2v2h2v-2zm0-6h-2v4h2v-4z"
              />
            </svg>
          </div>
          <div
            class="MuiAlert-message css-acap47-MuiAlert-message"
          >
            <div
              class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom MuiAlertTitle-root css-yslrua-MuiTypography-root-MuiAlertTitle-root"
            >
              Warning
            </div>
            Content
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 MuiAlert-root MuiAlert-standardWarning MuiAlert-standard css-rob5aa-MuiPaper-root-MuiAlert-root"
        role="alert"
      >
        <div
          class="MuiAlert-icon css-1ytlwq5-MuiAlert-icon"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
            data-testid="ReportProblemOutlinedIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 5.99L19.53 19H4.47L12 5.99M12 2L1 21h22L12 2zm1 14h-2v2h2v-2zm0-6h-2v4h2v-4z"
            />
          </svg>
        </div>
        <div
          class="MuiAlert-message css-acap47-MuiAlert-message"
        >
          <div
            class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom MuiAlertTitle-root css-yslrua-MuiTypography-root-MuiAlertTitle-root"
          >
            Warning
          </div>
          Content
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
